src/bootstrap/src/core/build_steps/format.rs

```diff
@@ -142,14 +142,17 @@ pub fn format(build: &Builder<'_>, check: bool, paths: &[PathBuf]) {
         };
         if in_working_tree {
             let untracked_paths_output = output(
-                build.config.git().arg("status").arg("--porcelain").arg("--untracked-files=normal"),
+                build
+                    .config
+                    .git()
+                    .arg("status")
+                    .arg("--porcelain")
+                    .arg("-z")
+                    .arg("--untracked-files=normal"),
+            );
+            let untracked_paths = untracked_paths_output.split_terminator('\0').filter_map(
+                |entry| entry.strip_prefix("?? "), /* returns None if the prefix doesn't match */
```

Comment:
a nit suggestion:

```suggestion
                |entry| entry.strip_prefix("?? "), // returns None if the prefix doesn't match
```

Reply:
Looks good, I missed that when the formatter split it over a few lines.